### PR TITLE
[WIP] Add config for Cirrus CI

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  test_script: mvn install -DskipTests -B -T 1C -P ci
+  docker_script: docker/build-local.sh
+
+maven_specific_modules_task:
+  container:
+    image: maven:3-jdk-8
+  maven_cache:
+    folder: ~/.m2
+  env:
+    matrix:
```
Comment (findepi) [inline .cirrus.yml]: is it "enumerate" or "multiple"? (71496aa777ef036896617f1943fb4b0c86932338)

File: .cirrus.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+maven_presto_tests_ci_task:
+  container:
+    image: maven:3-jdk-8
+  maven_cache:
+    folder: ~/.m2
+  install_script: mvn install $MAVEN_FAST_INSTALL -pl presto-tests -am
+  test_script: mvn test $MAVEN_SKIP_CHECKS_AND_DOCS -B -pl presto-tests -P ci-only
+
+maven_other_modules_task:
```
Comment (findepi) [inline .cirrus.yml]: I don't see any dockerized builds here. How are they handled?

File: .cirrus.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    image: node:latest
+  check_script: presto-main/bin/check_webui.sh
+
+maven_checks_task:
+  container:
+    image: maven:3-jdk-8
+  maven_cache:
+    folder: ~/.m2
+  test_script: mvn install -DskipTests -B -T 1C -P ci
+  docker_script: docker/build-local.sh
```
Comment (findepi) [inline .cirrus.yml]: How do they know `docker_script` is a name of executable and that it needs to run after `test_script`?

Comment (electrum) [reply .cirrus.yml]: See https://cirrus-ci.org/guide/writing-tasks/#script-instruction

It doesn't seem to explicitly say that they run in order, but it's clear from their examples that it works that way. We could send a PR to update the docs to say.

File: .cirrus.yml (hunk 2 above)
Comment (electrum) [reply .cirrus.yml]: I haven't got that far yet. When I emailed them previously, they said to look at

https://cirrus-ci.org/guide/writing-tasks/#additional-containers
https://cirrus-ci.org/guide/docker-builder-vm/

File: .cirrus.yml (hunk 1 above)
Comment (electrum) [reply .cirrus.yml]: See https://cirrus-ci.org/guide/writing-tasks/#matrix-modification

File: .cirrus.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    image: node:latest
+  check_script: presto-main/bin/check_webui.sh
+
+maven_checks_task:
+  ec2_instance:
+    image: ami-08d489468314a58df
+    type: m5.xlarge
+    region: us-east-1
+  container:
+    image: maven:3-jdk-8
```
Comment (fkorotkov) [inline .cirrus.yml]: You probably want to remove `container` from both of the tasks. This is the reason why `aws_credentials` got erased. Will try to make parser a bit smarter and throw on such cases when the instance is overridden on the same yaml level.

**Pull request closed without merging**